### ISSUE TITLE
Share room doesn't have a matrix option

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#6909]
Someone asked for an alternative alias to get into a room, and my first instinct was to click the share button. Would be nice to have this instinct yield results :)

**Comment by t3chguy** [comment]
I've thought about this, got any UX ideas?

One I had was to use the `Create or Reuse chat dialog` for you to select/make a 1:1 with someone to send it them
Another was it could launch the forwarding UX
Another was it'd just spit it into your composer, though for things like Room Sharing this'd be not useful at all

**Comment by turt2live** [comment]
In this case I was more interested in getting the link to the room than sending it to a private chat or something.

Considering it's a 30 second mockup, how does this look?
![image](https://user-images.githubusercontent.com/1190097/41559574-a605b8d8-7301-11e8-8a05-ffa4d51147a1.png)

**Comment by t3chguy** [comment]
oh I misunderstood from 

> Share room doesn't have a matrix option

**Comment by t3chguy** [comment]
the annoying thing there is the tickboxes cannot be used in conjunction, permalinks have room ids

**Comment by turt2live** [comment]
Permalinks can also have aliases, I've seen and used them before.

**Comment by t3chguy** [comment]
fair enough

**Comment by t3chguy** [comment]
Well I know one client that doesn't support that yet :P (matrix-static)

**Comment by lampholder** [comment]
/me observes that a permalink with an alias in it is not a permalink (since aliases are not permanent).

I threw together a quick and flawed mockup of a UX that tries to avoid conflating those concepts:

![new 3](https://user-images.githubusercontent.com/1922197/41847128-dd859408-7870-11e8-9e7d-ee3ac8a078fc.png)
![new 4](https://user-images.githubusercontent.com/1922197/41847130-de607366-7870-11e8-9c4c-ff430106dfa9.png)
![new 5](https://user-images.githubusercontent.com/1922197/41847134-df6ad54e-7870-11e8-97c5-820ba33df578.png)

Things I like:
- it shows that links with aliases in them are something other than a permalink
- you can find the alias you want to use quite easily

Things I don't like:
- more clicking
- less horizontal width in which to fit the URL

**Comment by t3chguy** [comment]
looks fancy tho :D

**Comment by turt2live** [comment]
This kinda feels like it took a turn towards being strongly related to this: https://github.com/vector-im/riot-web/issues/2925

Mockup looks good :) It probably shouldn't recommend `[ ] Link to most recent message` on aliases though if the intention is to differentiate them.

Also, Riot should absolutely pillify the matrix.to alias link if it doesn't already.

**Comment by MTRNord** [comment]
@lampholder isn't a ID server specific? In that case it could make a lot of misunderstandings having it as a option. Some might missunderstand it and give it too people who arent on their server which causes a lot of "why doesnt the permalink never work" questions on the riot room :/

**Comment by t3chguy** [comment]
@MTRNord yes. But it's the only permanent room identifier. Room aliases can change without control of the room.

**Comment by MTRNord** [comment]
@t3chguy I agree on that part. But I feel it needs an explenation that Room IDs are not federation wide

**Comment by turt2live** [comment]
@lampholder is the mockup you've provided (alongside it's dislikes) what you'd like to see implemented? or did you have another plan?

(also, I think this may have accidentally landed on the FiP board - I don't know how much a p2 feature is likely to affect a user's first experience of riot)

**Comment by lampholder** [comment]
> (also, I think this may have accidentally landed on the FiP board - I don't know how much a p2 feature is likely to affect a user's first experience of riot)

It ended up on the list because sharing links people can use to get onto the platform is a good way to encourage growth (so sharing a link mightn't be in your first 5 minutes of using the thing, but _clicking_ a shared link might).

Perhaps it should wait until https://github.com/vector-im/riot-web/issues/2925 is fixed, though?

**Comment by turt2live** [comment]
Clicking a shared link is more of a #2925 problem, which is also on the FiP board (yaaaaaaaaaaay).

**Comment by ara4n** [comment]
i am really surprised that this is on the FiP board; wanting to pick an alternative alias when sharing a link to a room is not exactly something a user is going to trip over in their first 24h on the app.   Meanwhile, all the issues with the solution here point back to #2925, so let's (at last!!!) fix that first, which should then make the problem here trivial to solve.  I'm going to de-FiP it meanwhile.
